presto-main/src/test/java/com/facebook/presto/sql/query/TestSubqueries.java

```diff
@@ -75,18 +98,19 @@ public void testCorrelatedSubqueriesWithLimit()
         assertions.assertQuery(
```

Comment:
I'm seeing that for other SQL features we verify plans and results in separate tests. E.g. `TestLogicalPlanner` verifies query plans and `AbstractTestQueries` verifies correctness of the results. Is there any particular reason to combine these verifications in a single test for correlated subqueries? Would it be clearer to have 2 separate tests? 

Also, some queries in this test now verify both results and plans, but some continue to verify only results. Shouldn't all queries be tested *at the same depth*?

Reply:
Good point. I think planner tests (e.g: `TestLogicalPlanner`) and tests in `com.facebook.presto.sql.query` are slowly converging. I think the we could categorize the tests as follows:
* Tests that use `QueryAssertions` are more of an "integration" tests that validate planner and engine end to end. They allow to write really convenient one-liner tests using SQL queries. Usually this is fine as with such tests we care mostly about capabilities of the engine (e.g: if it can execute given query).

* Tests that use `BasePlanTest` are more of an "unit/integration" tests of planner itself. They are less convenient to write as you have to write a pattern and make sure that the pattern doesn't break easily. I think those tests verify that particular rewrite triggered but assume that query already returns correct results.

In `TestSubqueries` primarily I want to ensure that query can execute but I also want to make sure that correct rewrite fired.

> Is there any particular reason to combine these verifications in a single test for correlated subqueries? Would it be clearer to have 2 separate tests?

I thought about it. I think dual tests would result in copy-pasting queries to two different classes. I'm also not doing full blown plan pattern assertion, but just looking where the aggregation ended up. Therefore I think in current form it's more readable.

However, maybe we should merge `BasePlanTest` and `QueryAssertions` since they both essentially are toolsets for testing planner mostly? What do you think @martint ?